### ISSUE TITLE
Standalone app doesn't work

### ISSUE THREAD

**Issue by shaman-apprentice** [MaibornWolff/codecharta#2823]
# Bug

## Expected Behavior
Given a new version is released
When downloading the assets
Then we get executable for Windows, Mac and Linux.

## Actual Behavior
See [ci output of job "Build visualization"](https://github.com/MaibornWolff/codecharta/runs/6468255242?check_suite_focus=true) and available assets of [release 1.96.0](https://github.com/MaibornWolff/codecharta/releases/tag/1.96.0):

- Windows version is not build due to missing `wine` in build image
- Mac x64 is not build due to a bug in [nwjs-builder-phoenix](https://github.com/evshiron/nwjs-builder-phoenix); maybe related to [this bug](https://github.com/evshiron/nwjs-builder-phoenix/issues/106) and [this bug](https://github.com/evshiron/nwjs-builder-phoenix/issues/139)
- When I download Linux x64 version and want to run it in my Ubuntu 20.04 WSL I get the following error: `(codecharta-visualization:293): Pango-ERROR **: 17:54:03.066: Harfbuzz version too old (1.2.1)`

## Open questions
- Do we need any standalone version and if so do we need Windows, Mac and Linux? -> erstmal ja
- For Linux: Is it the user responsibility to have the right version of `Harfbuzz` or out responsibility to use a not outdated version?
- **nwjs-builder-phoenix isn't maintained anymore. Do we want to migrate away from it?**

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json&file=codecharta_analysis.cc.json): yes
- CodeCharta Version: 1.96.0
- OS: Windows, Mac, Linux

**Comment by IhsenBouallegue** [comment]
If I understood correctly, we need to fix the standalone builds.
But in this issue it was mentioned that we need to remove them and keep the web version (which works on all platforms) https://github.com/MaibornWolff/codecharta/issues/2573.
I don't see a lot of benefit from the windows linux and mac versions. It's a lot more maintainability for little benefit.

**Comment by BridgeAR** [comment]
We might have different understandings when it comes to this: #2573 was about simplifying the build process for the standalone version. We used to have two types of builds: one for the web that caused different javascript code to be executed than in the standalone version. Now both builds are identical but we still have to be able to execute the standalone version properly.

So if I understood this issue correct it's only about not being able to properly start the standalone version at the moment and that should definitely be the case.

**Comment by IhsenBouallegue** [comment]
> So if I understood this issue correct it's only about not being able to properly start the standalone version at the moment and that should definitely be the case.

I see. Thanks for the clarification @BridgeAR.
And why do we have other standalone versions than web in the first place? (Mac, Windows, Linux…) are they still relevant? Shouldn't be web enough for all platforms?

**Comment by BridgeAR** [comment]
We require a desktop application that runs on any OS. The reason is mainly that Internet is not always available.

**Comment by BridgeAR** [comment]
It is correct, that it works by loading the main file in a browser. Some computers that CC should run upon might not have a browser though. We just have to make sure we need nothing on a computer than CC to be run properly. What tool is used for it does not matter. Let's please look into what has broken in nw.js soon. It could be a breaking change in a recent version (that's what I would verify first).

**Comment by luelista** [comment]
Okay, the fix for macOS worked, for Windows it still got problems. The Github Action reports an exit code = 123 error for rcedit.exe

**Comment by luelista** [comment]
Ich habe den nwjs Build unter Windows untersucht mit folgendem Ergebnis: Es läuft aktuell nicht, da der nwjs-builder-phoenix das Modul node-rcedit in einer sehr alten Version referenziert, die nur die 32-bit Version der "rcedit.exe" mitbringt. Daher läuft es in unserer 64-bit Buildpipeline nicht. Ein 32-bit kompatibles Wine dort zu installieren ist nicht trivial. Sinnvoller wäre es, die neuere node-rcedit zu nutzen, die unter 64-bit läuft. Mit folgendem snippet in der package.json kann ich zwar eine neuere node-rcedit erzwingen, allerdings läuft dann der builder nicht mehr durch, da sich das API von node-rcedit geändert hat:

```
"overrides": {
  "nwjs-builder-phoenix": {
    "rcedit": "^3.0.1"
  }
}
```

Wir müssten also unsere eigene, angepasste Version von nwjs-builder-phoenix pflegen.
 
Stattdessen würde ich empfehlen, erstmal das alternative Build-Tool nw-builder (https://github.com/nwutils/nw-builder) zu testen. Das ist zwar nur die zweite Wahl laut Doku (https://docs.nwjs.io/en/latest/For%20Users/Package%20and%20Distribute/), aber dafür deutlich aktueller (letztes Update 2022 statt 2018) - vielleicht ist die Doku veraltet. Es scheint auch ein sehr ähnliches Featureset zu haben.

**Comment by BridgeAR** [comment]
+1 on switching to nw-builder.

**Comment by phanlezz** [comment]
Fixed by #2881

**Comment by phanlezz** [comment]
Currently the web package and the global npm install don't work of the visualization part.

**Comment by ce-bo** [comment]
Would you please provide some More error details?

**Comment by phanlezz** [comment]
> Would you please provide some More error details?

After installing via `npm i -g codecharta-virtualization` and running it `codecharta-virtualization` under mac, only the command `nw .` pops up in the command line and process is finished. Nothing else happens.

The zip file of the -web version can't be extracted, because it is corrupted. Im already working on this second problem.

**Comment by phanlezz** [comment]
Closed via #2934, standalone applications should work now. There is still an issue with the MacOS npm-install version: #2967
